src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-update-firmware-node.ts

```diff
@@ -0,0 +1,444 @@
+                      (this._updateProgressMessage.sent_fragments * 100) /
+                      this._updateProgressMessage.total_fragments
+                    ).toFixed(2),
+                  }
+                )}
+              </p>
+              <ha-bar
+                min="0"
+                .value=${this._updateProgressMessage.sent_fragments}
+                .max=${this._updateProgressMessage.total_fragments}
+              >
+              </ha-bar>
+              <p>
+                ${this.hass.localize(
+                  "ui.panel.config.zwave_js.update_firmware.close",
+                  {
+                    device: this._deviceName,
+                  }
+                )}
+              </p>
+              ${abortFirmwareUpdateButton}
+            `
+          : html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${status === "ok" ? mdiCheckCircle : mdiCloseCircle}
+                  .class=${status}
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      `ui.panel.config.zwave_js.update_firmware.finished_status.${status}`,
+                      {
+                        device: this._deviceName,
+                        message: this.hass.localize(
+                          `ui.panel.config.zwave_js.update_firmware.finished_status.${
+                            FirmwareUpdateStatus[
+                              this._updateFinishedMessage!.status
+                            ]
+                          }`
+                        ),
+                      }
+                    )}
+                  </p>
+                </div>
+              </div>
+              <p>
+                ${this.hass.localize(
+                  "ui.panel.config.zwave_js.update_firmware.finished_status.try_again"
+                )}
+              </p>
+              ${beginFirmwareUpdateHTML}
+            `}
+      </ha-dialog>
+    `;
+  }
+
+  private async _fetchData(): Promise<void> {
+    let nodeStatus: ZWaveJSNodeStatus;
+    [this._firmwareUpdateCapabilities, nodeStatus, this._updateInProgress] =
+      await Promise.all([
+        fetchZwaveNodeFirmwareUpdateCapabilities(this.hass, this.device!.id),
+        fetchZwaveNodeStatus(this.hass, this.device!.id),
+        fetchZwaveNodeFirmwareUpdateProgress(this.hass, this.device!.id),
+      ]);
+    this._nodeStatus = nodeStatus.status;
+    if (this._updateInProgress) {
+      this._subscribeNodeFirmwareUpdate();
+    }
+  }
+
+  private async _beginFirmwareUpdate(): Promise<void> {
+    this._uploading = true;
+    this._updateProgressMessage = this._updateFinishedMessage = undefined;
+    try {
+      this._subscribeNodeFirmwareUpdate();
+      await uploadFirmware(
+        this.hass,
+        this.device!.id,
+        this._firmwareFile!,
+        this._target
+      );
+      this._updateInProgress = true;
```

Comment:
Does upload a firmware file also starts the update? The function name doesnt reflect that right now, I think that should be made clear.